Sources_v3/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -302,6 +312,21 @@ extension _ChatMessage {
         localState = dto.localMessageState
         
         isFlaggedByCurrentUser = dto.flaggedBy != nil
+        
+        latestReactions = Set(
+            dto
+                .reactions
+                .sorted(by: { $0.updatedAt > $1.updatedAt })
+                .prefix(.maxNumberOfLatestReactions)
+                .map { $0.asModel() }
+        )
+        
+        currentUserReactions = Set(
+            dto
+                .reactions
+                .filter { $0.user.isCurrentUser }
+                .map { $0.asModel() }
+        )
```

Comment:
Same here. Better to use fetch request + `CurrentUserDTO.user`

Reply:
fixed in 8ac7bb42aac85e9e420e5cfa0450d6536bfbda46